compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -1698,27 +1712,70 @@ public void visit(BLangTupleVariableDef bLangTupleVariableDef) {
     @Override
     public void visit(BLangRecordVariable bLangRecordVariable) {
         analyzeNode(bLangRecordVariable.typeNode, env);
+        if (bLangRecordVariable.expr != null) {
+            this.currDependentSymbol.push(bLangRecordVariable.symbol);
+            analyzeNode(bLangRecordVariable.expr, env);
+            this.currDependentSymbol.pop();
+            return;
+        } else {
```

Comment:
Since we are returning in the if block, can move the logic out and remove the else block.